### PR TITLE
implemented todo app

### PR DESCRIPTION
**Pull request by HashirMuhammad**

### CODE REVIEW

File: docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+services:
+
+  database:
+    image: mysql:latest
+    container_name: blog_db
+    restart: always
+    ports:
+      - "3306:3306"
+    environment:
+      MYSQL_DATABASE: blog
```
Comment (zunnorainzaheer) [inline docker-compose/docker-compose.yml]: why the db name is blog? it should be todo

File: docker-compose/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      MYSQL_ROOT_PASSWORD: admin
+  web:
+    image: nginx
```
Comment (zunnorainzaheer) [inline docker-compose/docker-compose.yml]: why nginx service is in this file?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"io"
+	"net/http"
+	"strconv"
+
+	"encoding/json"
```
Comment (zunnorainzaheer) [inline main.go]: all dependencies that are part of go library should be before the blank line

File: main.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+	http.ListenAndServe(":8000", handler)
+
+	if err != nil {
+		return
+	}
+
+}
+
+type n struct {
```
Comment (zunnorainzaheer) [inline main.go]: what is n?  plz give it a meaningful name.

File: main.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+func main() {
+	db, err := gorm.Open("mysql", "root:root@/todolist?charset=utf8&parseTime=True&loc=Local")
+	if err != nil {
+		panic(err)
+	}
+
+	defer db.Close()
+
+	//db.Debug().DropTableIfExists(&TodoItemModel{})
+	db.Debug().AutoMigrate(&TodoItemModel{})
```
Comment (zunnorainzaheer) [inline main.go]: don't use these functions. ask me for a session on this as well

**Review by zunnorainzaheer — CHANGES_REQUESTED** [review]
you have done all the codein one single file. start looking at package in golang

File: docker-compose/docker-compose.yml (hunk 1 above)
Comment (HashirMuhammad) [reply docker-compose/docker-compose.yml]: it was just an example to practice. it was not with todo app.

File: docker-compose/docker-compose.yml (hunk 2 above)
Comment (HashirMuhammad) [reply docker-compose/docker-compose.yml]: ok i have removed

File: main.go (hunk 3 above)
Comment (HashirMuhammad) [reply main.go]: ok

**Pull request merged**